SDK v2 migration guide.md

```diff
@@ -0,0 +1,235 @@
+  
+
+## Frequently Asked Questions
+
+Question:
+> What do I gain by upgrading to the 2.X.X release?
+
+Answer:
+> You get a smaller set of dependencies which makes for a lighter SDK overall. You also get a more concise and clear API surface. 
+> Lastly, and most importantly, you get an SDK that is decoupled from a particular IoT hub and Device Provisioning 
+> Service public certificate. This makes these versions more future proof since they aren't tied to a certificate
+> that will be changed within a few years and may be changed again beyond then. 
+
+Question:
+> Will the 1.X.X releases still be supported in any way?
+
+Answer:
+> We will continue to support the long term support releases of the 1.X.X SDK for their lifespans, but we will not bring 
+> newer features to the 1.X.X SDK. Users who want access to the upcoming features are encouraged to upgrade to the 2.X.X SDK.
+
+Question:
+> After upgrading, when I try to open my client, I get an exception like:
+> ```
+> TODO: add exception
+> ```
+
+Answer:
+> TODO: add solution
+
+Question:
+> If the SDK now reads certificates from my device's trusted root certification authorities certificate store, does it also read any private keys that I have installed on my device?
+
+Answer:
+TODO: does this apply to c#
+> No. This SDK only reads the public certificates from your device's trusted root certification authorities certificate store. 
+>
+> For x509 authenticated connections, you will need to construct the SSLContext with your trusted certificates and private keys and provide it to the SDK.
+> See [this sample](./iothub/device/samples/how%20to%20guides/X509DeviceCertWithChainSample/) for an example of how to construct an SSLContext instance with your public certificates and private keys.
+
+Question:
+> Can I still upload files to Azure Storage using this SDK now that deviceClient.uploadToBlobAsync() has been removed? (TODO: does this apply to c#?)
+
+Answer:
+> Yes, you will still be able to upload files to Azure Storage after upgrading. 
+>
+> This SDK allows you to get the necessary credentials to upload your files to Azure Storage, but you will need to bring in the Azure Storage SDK as a dependency to do the actual file upload step. 
+> 
+> For an example of how to do file upload after upgrading, see [this sample](./iothub/device/samples/getting%20started/FileUploadSample/).
+
+Question:
+> I was using a deprecated API that was removed in the 2.X.X upgrade, what should I do?
+
+Answer:
+> The deprecated API in the 1.X.X version documents which API you should use instead of the deprecated API. This guide
+also contains a mapping from v1 API to equivalent v2 API that should tell you which v2 API to use.
+
+Question:
+> After upgrading, some of my catch statements no longer work because the API I was using no longer declares that it throws that exception. Do I still need to catch something there?
```

Comment:
```suggestion
> After upgrading, some of catch handlers no longer work because the API I was using no longer declares that it throws that exception. Do I still need to catch something there?
```